selectors/rhs.jsx

```diff
@@ -41,3 +43,43 @@ export const getSelectedPost = createSelector(
         };
     }
 );
+
+export function makeGetCurrentUsersLatestPost(channelId, rootId) {
+    return createSelector(
+        getCurrentUserId,
+        (state) => state.entities.posts.postsInChannel[channelId] || [],
+        (state) => (id) => getPost(state, id),
```

Comment:
This is going to be returning a new function each time the selector is called, which will also prevent memoization.

It may be easier to just not use a selector here since the overhead and complexity of using a selector might not be worth it when it's unlikely the user will likely go to edit their post more than once. It may actually be slightly worse overall performance to use a selector since it's making a new function each time the channelId or rootId change when the user may not even use this hotkey.

Reply:
@hmhealey  What would like to do here? I can pull `(id) => getPost(state, id)` out fixing the memoizing issue.

Reply:
@hmhealey What would you like me to do here?

Reply:
I can move it to the same place as `makeOnEditLatestPost` as a util function instead, since this is the only place its used. That way it wont be used in `mapStateToProps`.

Reply:
That way it'll only do all this work when the user wants to edit their latest post.

Reply:
I think it might still be called whenever mapStateToProps is called for a new thread (ie a different root_id or channel_id) since that'll cause call makeOnEditLatestPost which'll call this in turn. That'll still create a new function here and blow away memoization, but we want that anyway